### ISSUE TITLE
How do I use CMake without having "compiler" in the ConanFile settings?

### ISSUE THREAD

**Issue by cleeland** [conan-io/conan#1438]
As I understand it, the ConanFile settings determine what facets of a build are compared to determine equivalence.  So, the default setting of `"os","build_type","arch","compiler"` means that if the OS, build_type, architecture or compiler changes, then conan will not match and will require a different build.

In my specific case, I am packaging a binary executable, not a library. To the user of the executable it does not matter what compiler was used--or it should not.  If I remove `"compiler"` from the list above, though, conan's cmake wrapper complains that it needs a compiler setting, and will not build.

How do I vary these two things independently? For a binary, I really only need "arch" and "os".  For libraries, it's common to compile using the oldest compiler that works in order to give the widest compatibility, so this isn't just a binary thing.

**Comment by memsharded** [comment]
> As I understand it, the ConanFile settings determine what facets of a build are compared to determine equivalence. So, the default setting of "os","build_type","arch","compiler" means that if the OS, build_type, architecture or compiler changes, then conan will not match and will require a different build.

Exactly that. Together with the ``options`` and the ``requires``, that define the package ID, if some value change, it results in a new ID, than require a different build.

What you are looking for is the ``package_id()`` method. It is used to define the "output" or the package ID. If your final executable is valid for different settings you can define so:

```python
def package_id(self):
    self.info.settings.compiler = "Any"
    self.info.settings.compiler.version = "Any"
    # etc, also to change options and requires
```

In this way you can both define settings inputs without having them affect the binary ID, then not requiring to build a new binary. Please have a look to: http://docs.conan.io/en/latest/packaging/package_info.html#package-binary-compatibility and

**Comment by cleeland** [comment]
Just so you know, I was actually looking at http://conanio.readthedocs.io/en/latest/howtos/define_abi_compatibility.html and playing with `package_id()` in an attempt to suss this out on my own.  I tried many things, but nothing in the ABI compatibility document suggested that `Any` was a valid setting.  It's briefly mentioned in the link you provided under "multi-configuration".  The link I referenced is what's referenced by [the documentation for `package_id()`](http://docs.conan.io/en/latest/reference/conanfile/methods.html#package-id).

I will try to figure out some suggestions for the documentation so that I would have found this without having to ask the question.
